lib/client/base.js

```diff
@@ -745,9 +745,16 @@
   HappnClient.prototype.reconnect = function (options) {
     var _this = this;
 
+    if (!this.reconnectAttempt) this.reconnectAttempt = 0;
+    var reconnectAttempt = ++this.reconnectAttempt;
+
     _this.authenticate(function (e) {
+      if (_this.reconnectAttempt != reconnectAttempt) return; // this is not the last attempt, ignore it
```

Comment:
I am not sure about this. This happens if request 1 is timing out and we have a reconnect. Request 2 is triggered before 1 times out. 2 is successful, but 1's timeout causes 3.